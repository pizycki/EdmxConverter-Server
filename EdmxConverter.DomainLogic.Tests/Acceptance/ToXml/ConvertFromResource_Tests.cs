using ApprovalTests;
using ApprovalTests.Namers;
using ApprovalTests.Reporters;
using EdmxConverter.DomainLogic.Service;
using EdmxConverter.DomainLogic.Tests.Properties;
using Xunit;

namespace EdmxConverter.DomainLogic.Tests.Acceptance.ToXml
{
    public class ConvertFromResource_Tests
    {
        [Fact]
        [UseReporter(typeof(DiffReporter))]
        [UseApprovalSubdirectory("Approved")]
        public void convert_from_plain_resource()
        {
            var resourceValue = Resources.SampleResourceEdmx;
            var resourceEdmx = new ResourceEdmx(resourceValue);
            ConvertToXml.FromResource(resourceEdmx)
                        .IfSucc(xml => Approvals.Verify(xml.ToString()));
        }
    }
}
