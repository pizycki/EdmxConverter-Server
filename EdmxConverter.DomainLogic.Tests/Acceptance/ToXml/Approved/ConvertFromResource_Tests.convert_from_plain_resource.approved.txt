<Edmx Version="3.0" xmlns="http://schemas.microsoft.com/ado/2009/11/edmx">
  <Runtime>
    <ConceptualModels>
      <Schema Namespace="SampleDatabase.SampleDbContext" Alias="Self" annotation:UseStrongSpatialTypes="false" xmlns:annotation="http://schemas.microsoft.com/ado/2009/02/edm/annotation" xmlns:customannotation="http://schemas.microsoft.com/ado/2013/11/edm/customannotation" xmlns="http://schemas.microsoft.com/ado/2009/11/edm">
        <EntityType Name="Foo" customannotation:ClrType="SampleDatabase.SampleDbContext.Entities.Foo, SampleDatabase, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null">
          <Key>
            <PropertyRef Name="Id" />
          </Key>
          <Property Name="Id" Type="Guid" Nullable="false" />
          <Property Name="Name" Type="String" MaxLength="Max" FixedLength="false" Unicode="true" />
        </EntityType>
        <EntityContainer Name="SampleDbContext" customannotation:UseClrTypes="true">
          <EntitySet Name="Foos" EntityType="Self.Foo" />
        </EntityContainer>
      </Schema>
    </ConceptualModels>
    <Mappings>
      <Mapping Space="C-S" xmlns="http://schemas.microsoft.com/ado/2009/11/mapping/cs">
        <EntityContainerMapping StorageEntityContainer="CodeFirstDatabase" CdmEntityContainer="SampleDbContext">
          <EntitySetMapping Name="Foos">
            <EntityTypeMapping TypeName="SampleDatabase.SampleDbContext.Foo">
              <MappingFragment StoreEntitySet="Foo">
                <ScalarProperty Name="Id" ColumnName="Id" />
                <ScalarProperty Name="Name" ColumnName="Name" />
              </MappingFragment>
            </EntityTypeMapping>
          </EntitySetMapping>
        </EntityContainerMapping>
      </Mapping>
    </Mappings>
    <StorageModels>
      <Schema Namespace="CodeFirstDatabaseSchema" Provider="System.Data.SqlClient" ProviderManifestToken="2012" Alias="Self" xmlns:customannotation="http://schemas.microsoft.com/ado/2013/11/edm/customannotation" xmlns="http://schemas.microsoft.com/ado/2009/11/edm/ssdl">
        <EntityType Name="Foo">
          <Key>
            <PropertyRef Name="Id" />
          </Key>
          <Property Name="Id" Type="uniqueidentifier" Nullable="false" />
          <Property Name="Name" Type="nvarchar(max)" Nullable="true" />
        </EntityType>
        <EntityContainer Name="CodeFirstDatabase">
          <EntitySet Name="Foo" EntityType="Self.Foo" Schema="dbo" Table="Foos" />
        </EntityContainer>
      </Schema>
    </StorageModels>
  </Runtime>
  <Designer>
    <Connection>
      <DesignerInfoPropertySet>
        <DesignerProperty Name="MetadataArtifactProcessing" Value="EmbedInOutputAssembly" />
      </DesignerInfoPropertySet>
    </Connection>
    <Options>
      <DesignerInfoPropertySet>
        <DesignerProperty Name="ValidateOnBuild" Value="False" />
        <DesignerProperty Name="CodeGenerationStrategy" Value="None" />
        <DesignerProperty Name="ProcessDependentTemplatesOnSave" Value="False" />
        <DesignerProperty Name="UseLegacyProvider" Value="False" />
      </DesignerInfoPropertySet>
    </Options>
    <Diagrams />
  </Designer>
</Edmx>